packages/rrweb/src/replay/index.ts

```diff
@@ -1322,6 +1322,31 @@ export class Replayer {
         }
         break;
       }
+      case IncrementalSource.Selection: {
+        const doc = this.iframe.contentDocument;
+
+        const ranges = d.ranges.map(
+          ({ start, startOffset, end, endOffset }) => {
+            const startContainer = this.mirror.getNode(start);
+            const endContainer = this.mirror.getNode(end);
+
+            if (!startContainer || !endContainer) return;
+
+            const result = new Range();
+
+            result.setStart(startContainer, startOffset);
+            result.setEnd(endContainer, endOffset);
+
+            return result;
+          },
+        );
+
+        const selection = doc?.getSelection();
+        selection?.removeAllRanges();
+
+        ranges.forEach((r) => r && selection?.addRange(r));
```

Comment:
```suggestion
        const selectionSet = new Set<Selection>();
        const ranges = d.ranges.map(
          ({ start, startOffset, end, endOffset }) => {
            const startContainer = this.mirror.getNode(start);
            const endContainer = this.mirror.getNode(end);

            if (!startContainer || !endContainer) return;

            const result = new Range();

            result.setStart(startContainer, startOffset);
            result.setEnd(endContainer, endOffset);
            const doc = startContainer.ownerDocument;
            const selection = doc?.getSelection();
            selection && selectionSet.add(selection);

            return {
              range: result,
              selection,
            };
          },
        );

        selectionSet.forEach((s) => s.removeAllRanges());

        ranges.forEach((r) => r && r.selection?.addRange(r.range));
```
One more issue is that if the recorded selection change is inside an IFrame element, the selection target should be changed to iframe's selection.